### ISSUE TITLE
When viewing resource details, boolean field in pivot table for a many-to-many relationship always shows "red dot"

### ISSUE THREAD

**Issue by haoyuguo** [laravel/nova-issues#332]
I have two models `App\Account` and `App\Stakeholder`.  They have a many-to-many relationship.

The pivot table `account_stakeholder` has a field called "is_account_owner", indicating whether the stakeholder is the owner of the account.  

Stakeholder resource is defined like this (Account resource not included here but pretty similar):

```
    // Nova/Stakeholder.php
    public function fields(Request $request)
    {
        return [
            BelongsToMany::make('Accounts', 'accounts')->fields(function() {
                return [ Boolean::make('is_account_owner') ];
            })->searchable(),
        ];
    }
```

An account can be attached to and detached from a stakeholder as expected when viewing resource details, the additional field `is_account_owner` does show up, and is updated corrected (when checked in pivot table in database directly).  But this boolean field always shows up as "red dot" false value, even it has been set to true (seen in database).

**Comment by megan-starr9** [comment]
P sure this is the same as #186

**Comment by loren138** [comment]
I'm unclear if #186 refers to trying to pull pivot columns all the way to the index or just on the details page.

When linking, the pivot columns show as blank on the details page even though values exist in the database:
![screen shot 2018-09-07 at 3 28 16 pm](https://user-images.githubusercontent.com/1224505/45239275-b0818080-b2b2-11e8-992e-940b665defde.png)

By enabling var_dump of queries, I was able to get all the queries run by http://localhost:8000/nova-api/products?search=&filters=&orderBy=&orderByDirection=desc&perPage=25&trashed=&page=1&viaResource=sorts&viaResourceId=2&viaRelationship=products&relationshipType=belongsToMany

The query never selects the other columns out of the pivot table.
`"select "products".*, "Sorts_ProductsList"."SortID" as "pivot_SortID", "Sorts_ProductsList"."ProductID" as "pivot_ProductID" from "products" inner join "Sorts_ProductsList" on "products"."ID" = "Sorts_ProductsList"."ProductID" where "Sorts_ProductsList"."SortID" = ? order by "Name" asc limit 6 offset 0"`
[pageResult.txt](https://github.com/laravel/nova-issues/files/2362274/pageResult.txt)

The update page for the attached relation also shows blank inputs for me.

I made a Nova page for the Joining table, which shows the data just fine (the bottom relation is the one that should be on that details page):
![screen shot 2018-09-07 at 3 34 08 pm](https://user-images.githubusercontent.com/1224505/45239529-9005f600-b2b3-11e8-9fd6-603f3e5090af.png)
